paimon-core/src/main/java/org/apache/paimon/rest/auth/DLFAuthProvider.java

```diff
@@ -0,0 +1,175 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.rest.auth;
+
+import org.apache.paimon.utils.FileIOUtils;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import static org.apache.paimon.rest.RESTObjectMapper.OBJECT_MAPPER;
+
+/** Auth provider for <b>Ali CLoud</b> DLF. */
+public class DLFAuthProvider implements AuthProvider {
+
+    public static final String DLF_HOST_HEADER_KEY = "Host";
+    public static final String DLF_AUTHORIZATION_HEADER_KEY = "Authorization";
+    public static final String DLF_DATA_MD5_HEX_HEADER_KEY = "x-dlf-data-md5-hex";
+    public static final String DLF_DATE_HEADER_KEY = "x-dlf-date";
+    public static final String DLF_SECURITY_TOKEN_HEADER_KEY = "x-dlf-security-token";
+    public static final String DLF_ACCESSKEY_ID_HEADER_KEY = "x-dlf-accesskey-id";
+    public static final double EXPIRED_FACTOR = 0.4;
+
+    private static final DateTimeFormatter DATE_FORMATTER =
+            DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss'Z'");
+
+    private final String tokenFilePath;
+
+    protected DLFToken token;
+    private final boolean keepRefreshed;
+    private Long expiresAtMillis;
+    private final Long tokenRefreshInMills;
+
+    public static DLFAuthProvider buildRefreshToken(
+            String tokenFilePath, Long tokenRefreshInMills) {
+        DLFToken token = readToken(tokenFilePath);
+        Long expiresAtMillis = getExpirationInMills(token.getExpiration());
+        return new DLFAuthProvider(
+                tokenFilePath, token, true, expiresAtMillis, tokenRefreshInMills);
+    }
+
+    public static DLFAuthProvider buildAKToken(
+            String accessKeyId, String accessKeySecret, String securityToken) {
+        DLFToken token = new DLFToken(accessKeyId, accessKeySecret, securityToken, null);
+        return new DLFAuthProvider(null, token, false, null, null);
+    }
+
+    public DLFAuthProvider(
+            String tokenFilePath,
+            DLFToken token,
+            boolean keepRefreshed,
+            Long expiresAtMillis,
+            Long tokenRefreshInMills) {
+        this.tokenFilePath = tokenFilePath;
+        this.token = token;
+        this.keepRefreshed = keepRefreshed;
+        this.expiresAtMillis = expiresAtMillis;
+        this.tokenRefreshInMills = tokenRefreshInMills;
+    }
+
+    @Override
+    public Map<String, String> header(
+            Map<String, String> baseHeader, RESTAuthParameter restAuthParameter) {
+        try {
+            String date = getDate();
+            String dataMd5Hex = DLFAuthSignature.md5Hex(restAuthParameter.data());
```

Comment:
if restAuthParameter.data() is empty, we can ignore the DLF_DATA_MD5_HEX_HEADER_KEY header